zipline/assets/assets.py

```diff
@@ -0,0 +1,543 @@
+            counter += 1
+
+    def spawn_asset(self, identifier, **kwargs):
+
+        # Check if the sid is declared
+        try:
+            kwargs['sid']
+            pass
+        except KeyError:
+            # Assign the identifier as the sid, if applicable
+            if isinstance(identifier, int):
+                kwargs['sid'] = identifier
+            # If the identifier is not a sid, assign one
+            else:
+                kwargs['sid'] = self._assign_sid(identifier)
+                # Update the metadata object with the new sid
+                self.insert_metadata(identifier=identifier, sid=kwargs['sid'])
+
+        # If the file_name is in the kwargs, it may be the symbol
+        try:
+            file_name = kwargs.pop('file_name')
+            try:
+                kwargs['symbol']
+            except KeyError:
+                kwargs['symbol'] = file_name
+        except KeyError:
+            pass
+
+        # If the identifier coming in was a string and there is no defined
+        # symbol yet, set the symbol to the incoming identifier
+        try:
+            kwargs['symbol']
+            pass
+        except KeyError:
+            if isinstance(identifier, string_types):
+                kwargs['symbol'] = identifier
+
+        # If the company_name is in the kwargs, it may be the asset_name
+        try:
+            company_name = kwargs.pop('company_name')
+            try:
+                kwargs['asset_name']
+            except KeyError:
+                kwargs['asset_name'] = company_name
+        except KeyError:
+            pass
+
+        # If dates are given as nanos, pop them
+        try:
+            kwargs['start_date'] = kwargs.pop('start_date_nano')
+        except KeyError:
+            pass
+        try:
+            kwargs['end_date'] = kwargs.pop('end_date_nano')
+        except KeyError:
+            pass
+        try:
+            kwargs['notice_date'] = kwargs.pop('notice_date_nano')
+        except KeyError:
+            pass
+        try:
+            kwargs['expiration_date'] = kwargs.pop('expiration_date_nano')
+        except KeyError:
+            pass
+
+        # Process dates
+        try:
+            kwargs['start_date'] = pd.Timestamp(self.trading_calendar.\
+                canonicalize_datetime(pd.Timestamp(kwargs['start_date'])))
+        except KeyError:
+            pass
+        try:
+            kwargs['end_date'] = pd.Timestamp(self.trading_calendar.\
+                canonicalize_datetime(pd.Timestamp(kwargs['end_date'])))
+        except KeyError:
+            pass
+        try:
+            kwargs['notice_date'] = pd.Timestamp(self.trading_calendar.\
+                canonicalize_datetime(pd.Timestamp(kwargs['notice_date'])))
+        except KeyError:
+            pass
+        try:
+            kwargs['expiration_date'] = pd.Timestamp(self.trading_calendar.\
+                canonicalize_datetime(pd.Timestamp(kwargs['expiration_date'])))
```

Comment:
@jfkirk does the expiration date have to be canonicalized? I would like to be able to use hours and minutes.